### ISSUE TITLE
Process of 'undefined'

### ISSUE THREAD

**Issue by lfantone** [z0mt3c/node-restify-validation#36]
Each time I pass a validation object to a route, for i.e:

```javascript
server.get({url: '/test/:name', validation: {queries: {age: {isRequired: true}}}}, function(req, res, next) {
  res.send(req.params);
  next();
});
```

I get the following error returned by the ***node-restify-validation*** module:

    {
      code: "InternalError",
      message: "Cannot read property 'process' of undefined"
    }

I've take a look at the _node-restify-validation/lib/index.js_ and saw that the `self` variable is the current server context instead of the modules one.
In order to get it work I've to re-create the wanted context:

```javascript
var self = {
  validation: require('./validation'),
  error: require('./error'),
  when: require('./conditions')
}
module.exports.validation = self.validation;
module.exports.error = self.error;
module.exports.when = self.when;
```

**notes :**
    Node  v0.12.7
    Restify v3.0.3
    Node Restify Validation v1.0.4

**Comment by gchauvet** [comment]
I've added a integration test between restify and node-restify-validation.
Please complete this IT with your usecase (I've used your route example as skeleton)

**Comment by gchauvet** [comment]
You patch is good but I need to fix #32 in the same time
